src/NuGet.Clients/NuGet.CommandLine/Commands/RestoreCommand.cs

```diff
@@ -227,10 +227,19 @@ private string GetSolutionDirectory(PackageRestoreInputs packageRestoreInputs)
 
         private void ReadSettings(PackageRestoreInputs packageRestoreInputs)
         {
+            var solutionDirectory = string.Empty;
+
             if (IsSolutionRestore(packageRestoreInputs))
             {
-                var solutionDirectory = GetSolutionDirectory(packageRestoreInputs);
+                solutionDirectory = GetSolutionDirectory(packageRestoreInputs);
+            }
+            else if (packageRestoreInputs.PackagesConfigFiles.Any())
```

Comment:
nit: Seems like this should already be working?

Reply:
If I just restore single `.csproj` file with package.config using `nuget.exe` then it not picking up nearest nuget.config due to `solutionDirectory ` is not set. This one os covering that edge case.

Reply:
Are you saying this is a bug?

It might be the wanted experience, I'd be hesitant to change that.

Reply:
Ok. Reverted this part.